charts/keycloak/templates/ingress.yaml

```diff
@@ -35,13 +35,13 @@ spec:
         paths:
           {{- range .paths }}
           - path: {{ . }}
-            {{- if $.Capabilities.APIVersions.Has "networking.k8s.io/v1/ingress" }}
+            {{- if $.Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" }}
             pathType: Prefix
             backend:
               service:
                 name: {{ include "keycloak.fullname" $ }}-http
                 port:
-                  name: {{ $ingress.servicePort }}
+                  {{ ternary "name" "number" (kindIs "string" $ingress.servicePort) }}: {{ $ingress.servicePort }}
```

Comment:
I don't see why we want to change this. It's supposed to be a string. The named port should be referenced.